clang/include/clang/AST/Type.h

```diff
@@ -2000,6 +2001,21 @@ class alignas(TypeAlignment) Type : public ExtQualsTypeCommonBase {
     unsigned NumExpansions;
   };
 
+  class CountAttributedTypeBitfields {
+    friend class CountAttributedType;
+
+    LLVM_PREFERRED_TYPE(TypeBitfields)
+    unsigned : NumTypeBits;
+
+    /// The limit is 15.
```

Comment:
Why is the limit 15? From what I can tell `NumTypeBits` is 22, so with `CountInBytes` and `OrNull` that's 24 bits, which makes 15 seem very arbitrary. How about a static assert that `sizeof(CountAttributedTypeBitfields)` is less than or equal to some maximum, if that's the goal here.

Reply:
Thanks! I added `static_assert` and removed the comment.